pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -145,67 +152,123 @@ func (p *InstanceTypeProvider) getAllInstanceTypes(ctx context.Context) ([]*ec2.
 		return true
 	}
-	if len(instanceTypeConstraints) != 0 && functional.ContainsString(instanceTypeConstraints, *instance.InstanceType) {
+	if len(instanceTypeConstraints) != 0 && functional.ContainsString(instanceTypeConstraints, instanceType.Name()) {
 		return true
 	}
 	return false
 }
 
 // isDefaultInstanceType returns true if the instance type provided conforms to the default instance type criteria
 // This function is used to make sure we launch instance types that are suited for general workloads
-func (p *InstanceTypeProvider) isDefaultInstanceType(instanceTypeInfo *packing.Instance) bool {
-	return instanceTypeInfo.FpgaInfo == nil &&
-		!*instanceTypeInfo.BareMetal &&
-		functional.HasAnyPrefix(*instanceTypeInfo.InstanceType,
+func (p *InstanceTypeProvider) isDefaultInstanceType(instanceType *InstanceType) bool {
+	return instanceType.FpgaInfo == nil &&
+		!*instanceType.BareMetal &&
+		functional.HasAnyPrefix(instanceType.Name(),
 			"m", "c", "r", "a", // Standard
 			"t3", "t4", // Burstable
 			"p", "inf", "g", // Accelerators
 		)
 }
 
-func (p *InstanceTypeProvider) isArchitectureSupported(architecture *string, instance *packing.Instance) bool {
-	return architecture == nil ||
-		functional.ContainsString(aws.StringValueSlice(instance.ProcessorInfo.SupportedArchitectures), *architecture)
+func (p *InstanceTypeProvider) isArchitectureSupported(architecture *string, instance *InstanceType) bool {
+	if architecture == nil {
+		return true
+	}
+	return functional.ContainsString(aws.StringValueSlice(instance.ProcessorInfo.SupportedArchitectures), *architecture)
 }
 
-func (p *InstanceTypeProvider) isCapacityTypeSupported(capacityType string, instance *packing.Instance) bool {
-	return capacityType == "" ||
-		functional.ContainsString(aws.StringValueSlice(instance.SupportedUsageClasses), capacityType)
+func (p *InstanceTypeProvider) isCapacityTypeSupported(capacityType string, instance *InstanceType) bool {
+	if capacityType == "" {
+		return true
+	}
+	return functional.ContainsString(aws.StringValueSlice(instance.SupportedUsageClasses), capacityType)
 }
 
-func (p *InstanceTypeProvider) isNvidiaGPUSupported(requests v1.ResourceList, instanceTypeInfo *packing.Instance) bool {
-	if _, ok := requests[resources.NvidiaGPU]; ok {
-		return instanceTypeInfo.GpuInfo != nil && *instanceTypeInfo.GpuInfo.Gpus[0].Manufacturer == "NVIDIA"
+func (p *InstanceTypeProvider) isNvidiaGPUSupported(requests v1.ResourceList, instanceType *InstanceType) bool {
+	if _, ok := requests[resources.NvidiaGPU]; !ok {
+		return true
 	}
-	return true
+	return !instanceType.NvidiaGPUs().IsZero()
+}
+func (p *InstanceTypeProvider) isAWSNeuronSupported(requests v1.ResourceList, instanceType *InstanceType) bool {
+	if _, ok := requests[resources.AWSNeuron]; !ok {
+		return true
+	}
+	return !instanceType.AWSNeurons().IsZero()
+}
+
+func (p *InstanceTypeProvider) isZonesSupported(zones []string, instanceType *InstanceType) bool {
```

Comment:
nit is -> are? Also name is a bit confusing regardless; IIUC this is more like "any/some zones supported", "is" or "are" are kinda ambiguous as to whether you mean all of them or just at least one.

Reply:
The naming of all of these is a bit complex, especially GPU (which is really semantically "has if needs"). Would love any high level naming recommendation you have for all of these.

Reply:
I think the `validate` change looked good.